operator/es_client.go

```diff
@@ -292,6 +300,14 @@ func (c *ESClient) waitForEmptyEsNode(ctx context.Context, pod *v1.Pod) error {
 					}
 				}
 				c.logger().Infof("Found %d remaining shards on %s/%s (%s)", remainingShards, pod.Namespace, pod.Name, podIP)
+
+				// make sure the IP is still excluded, this could have been updated in the meantime.
+				if remainingShards > 0 {
+					err = c.excludePodIP(pod)
```

Comment:
we could also abort the retry after some time. There could be valid reasons why ES wouldn't relocate all data away from the node. For example if no other node is eligible for taking the shards.

Reply:
(but personally, I prefer the repeating of excludePodIP here)